### PR TITLE
Support DDTree

### PR DESCRIPTION
**Pull request by Seunghui98**
## Description

This PR adds **DDTree** (tree-based speculative decoding) as a small, self-contained C++ core for nntrainer.

Speculative decoding amortizes the cost of an expensive model forward: a cheap *draft* proposes several next tokens and a single *target verify* forward accepts the longest prefix the target agrees with. DDTree proposes the candidates as a **tree** (rather than a single chain), so one verify forward evaluates many alternative continuations at once, raising the number of tokens accepted per step. Output stays lossless — the target verify recomputes true per-node logits, so draft quality affects only throughput, not correctness.

The core is **pure host logic**: it depends on the C++ standard library only, holds no KV cache, and has no `Tensor`/model dependency. From draft logits it builds the candidate tree, compiles it into verify input/position ids and an additive attention mask (with a sliding-window variant for models that mix full and sliding attention), follows the accepted root→leaf path after verification, and reorders the KV tail to the accepted nodes. A CPU or NPU runtime drives it by linking `libnntrainer`; the tree-construction and acceptance logic are reused as-is, not reimplemented per backend.

## Summary

- Add the `nntrainer/ddtree` leaf module (`nntrainer::ddtree`), standard-library only:
  - `buildTree` — best-first candidate-tree construction from draft logits.
  - `compile` — flatten the tree into verify input/position ids + an additive attention mask.
  - `makeSlidingMasks` (fp32 additive) / `makeSlidingVisibility` (0/1 bitmap) — sliding-window
    attention masks for models that interleave full and sliding-window attention.
  - `followVerified` — walk the longest accepted root→leaf path from a per-node argmax.
  - `compactTail` — model-agnostic raw-pointer KV tail reorder to the accepted nodes.
  - `argmaxRow` / `sampleGreedy` — temperature-0 sampling convenience helpers.
- Register the module so its sources flow into the meson and Android source lists, and install the public headers to `include/nntrainer` (added to the Debian and Tizen `-dev` packages).
- Add `unittest_ddtree` (21 tests) and Python-dumped golden vectors that lock the tree build (heap tie-break ordering and fp32/double numeric precision) bit-for-bit.
- Add module documentation (`nntrainer/utils/ddtree/README.md`).

Signed-off-by: SeungHui Lee <shsh1004.lee@samsung.com>

## Test

Built and ran the unit suite (`meson setup build -Denable-test=true && meson test -C build unittest_ddtree`):

```
[----------] 5 tests from DDTreeScaffold/DDTreeBuild ...
[ RUN      ] DDTreeBuild.MatchesPythonGolden
[       OK ] DDTreeBuild.MatchesPythonGolden (0 ms)
[ RUN      ] DDTreeSliding.VisibilityMatchesThresholdedMask
[       OK ] DDTreeSliding.VisibilityMatchesThresholdedMask (0 ms)
...
[==========] 21 tests from 7 test suites ran. (0 ms total)
[  PASSED  ] 21 tests.
```

- Self evaluation:
  1. Build test: [X]Passed [ ]Failed [ ]Skipped
  2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: SeungHui Lee <shsh1004.lee@samsung.com>

### CODE REVIEW

**Review by haehun — APPROVED** [review]
This PR implements ddtree using heaps from standard libraries and has unit test that compares its result to original Python code. LGTM.

**Review by jaemini-shin — APPROVED** [review]
LGTM!

**Review by dlwlzzero — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM
